src/main/java/software/amazon/awssdk/crt/s3/S3ResponseContext.java

```diff
@@ -0,0 +1,33 @@
+package software.amazon.awssdk.crt.s3;
+
+public class S3ResponseContext {
+    private final int errorCode;
+    private final int responseStatus;
+    private final byte[] errorPayload;
+    private final ChecksumAlgorithm checksumAlgorithm;
+    private final boolean didValidateChecksum;
+
+    public S3ResponseContext(final int errorCode, final int responseStatus, final byte[] errorPayload, final ChecksumAlgorithm checksumAlgorithm, final boolean didValidateChecksum) {
+        this.errorCode = errorCode;
+        this.responseStatus = responseStatus;
+        this.errorPayload = errorPayload;
+        this.checksumAlgorithm = checksumAlgorithm;
+        this.didValidateChecksum = didValidateChecksum;
+    }
+
+    public int getErrorCode () {
+        return this.errorCode;
+    }
+    public int getResponseStatus () {
+        return this.responseStatus;
+    }
+    public byte[] getErrorPayload () {
+        return this.errorPayload;
```

Comment:
we can get rid of errorPayload as its own member if we use CrtS3RuntimeException